# Average Wait Time
### Stochastic Calculation
What is the average wait time that a passenger would have to wait given:
1. Bus route
2. Inbound vs. Outbound
3. Timetable (Saturdays, Sundays, Weekdays)
4. A time period
5. Random arrival to the bus stop

In [ ]:
import datetime, numpy as np, pandas as pd

In this toy example, let's assume the following times represent minutes in a given hour. Thus in this example, we are looking at let's say 5 to 6 PM. There are three buses arriving at 5:20, 5:40, and 5:59 respectively. What is the average time that a passenger, arriving at random, would have to wait?

In [68]:
# Create the fake array
toy_example = [20, 40, 59]

# Instantiate i at 0 for the first element of the array of arrival minutes and the ranges as an empty array
ranges = []
i = 0

# 
while i < len(toy_example):
    range = np.arange(toy_example[i],-1,-1)
    ranges.append(range)
    
    i += 1

flattened_ranges = np.concatenate( ranges, axis=0 )

sum(flattened_ranges) / len(flattened_ranges)

22.950819672131146

For this single hour for this toy bus route, if buses come at hh:20:00, hh:40:00, and hh:59:00, the average wait time given a random arrival time at the stop in question is 22.95 minutes.